protocol/contracts/v2/TempleDebtToken.sol

```diff
@@ -145,17 +145,19 @@ contract TempleDebtToken is ITempleDebtToken, TempleElevatedAccess {
      * @notice Update the continuously compounding (base) interest rate of all debtors, from this block onwards.
      */
     function setBaseInterestRate(uint96 _rate) external override onlyElevatedAccess {
-        _checkpointBase(_compoundedBaseInterest());
+        // First checkpoint the base interest
+        _getBaseCache();
         baseRate = _rate;
         emit BaseInterestRateSet(_rate);
     }
 
     /**
      * @notice Update the continuously compounding (risk premium) interest rate for a given debtor, from this block onwards
      */
-    function setRiskPremiumInterestRate(address _debtor, uint64 _rate) external override onlyElevatedAccess {
+    function setRiskPremiumInterestRate(address _debtor, uint96 _rate) external override onlyElevatedAccess {
+        // First checkpoint the base interest & debtor risk premium interest
         Debtor storage debtor = debtors[_debtor];
-        _checkpointDebtor(debtor);
+        _getDebtorCache(_getBaseCache(), debtor);
```

Comment:
This is a functional change (bug fix). Previously the base debt wasn't checkpoint at this point, only the debtor info was checkpoint.

Now it correctly does both (see test updates which check this now)